modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -167,4 +167,63 @@ public static int hash(long key) {
 
         return hash(val);
     }
+
+    /**
+     * Constructs {@code long} from byte array.
+     *
+     * @param bytes Array of bytes.
+     * @param off Offset in {@code bytes} array.
+     * @return Long value.
+     */
+    public static long bytesToLong(byte[] bytes, int off) {
```

Comment:
As for me, this is not a great idea to write this code in java 11

Use this instead
```
static final VarHandle avh = MethodHandles.byteArrayViewVarHandle(long[].class, ByteOrder.BIG_ENDIAN);

byte[] longToBytes(long l) {
byte[] arr = new byte[8];
avh.set(arr, 0, context.value);
return arr;
}

long bytesToLong(byte[] val, int offset) {
return (long)avh.get(context.bytesValue, offset / 8 );
}

```

This is faster, because of use of intrinsics, and less error prone.

Reply:
```
Benchmark                                 Mode  Cnt   Score   Error  Units
BytesLongBenchmark.bytesToLongByteBuffer  avgt   10   4.173 ± 0.168  ns/op
BytesLongBenchmark.bytesToLongCycle       avgt   10   6.716 ± 0.137  ns/op
BytesLongBenchmark.bytesToLongUnroll      avgt   10   6.476 ± 0.166  ns/op
BytesLongBenchmark.bytesToLongVarHandle   avgt   10   4.879 ± 0.190  ns/op
BytesLongBenchmark.longToBytesByteBuffer  avgt   10   7.691 ± 0.245  ns/op
BytesLongBenchmark.longToBytesCycle       avgt   10  10.436 ± 0.386  ns/op
BytesLongBenchmark.longToBytesUnroll      avgt   10   9.716 ± 0.380  ns/op
BytesLongBenchmark.longToBytesVarHandle   avgt   10   7.779 ± 0.205  ns/op
```

Reply:
I would say that this is out of the scope of the current ticket, lets create a separate ticket with proper investigation and benchmarking.

Reply:
I can send you my benchmarks. Yes, I agree that difference is negligible. But I suppose, that ad-hoc is always worse than ready and battle-tested implementation.Also var handle use `_putLongUnaligned` and `_getLongUnaligned`.
 
By the way, why did you choose BIG_ENDIAN? As far as I remember, LITTLE_ENDIAN is used widely in Ignite (i.e. thin client protocol).